lib/ui/main.dart

```diff
@@ -132,6 +134,42 @@ class _MyHomePageState extends State<MyHomePage> {
     );
   }
 
+  AlertDialog buildDeleteAlertDialog(Alert currentAlert) {
+    return AlertDialog(
+      title: const Text("Delete Alert"),
+      titleTextStyle: const TextStyle(
+          fontWeight: FontWeight.bold, color: Colors.black, fontSize: 20),
+      actionsOverflowButtonSpacing: 20,
+      actions: [
+        ElevatedButton(
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+            style: const ButtonStyle(
+              backgroundColor: MaterialStatePropertyAll<Color>(Colors.grey),
+            ),
+            child: const Text("Cancel")),
+        ElevatedButton(
+          onPressed: () async {
+            await AlarmDatabase.instance.delete(currentAlert.id!).then(
```

Comment:
hmm the id field is a bit problematic, i wonder if we can have a way to avoid nullability on a primary key auto-increment field like this id..

Created a [ticket](https://www.notion.so/76f32de0991c4f0ca592d3ba9e175417?v=8f65456e0b9c4a2dacafbf964540a7f4&p=f319cb2a627940428ada5a16e1931224&pm=s) to track

Reply:
yea I haven't looked into the database part. Will learn :D